### PR TITLE
Propane Tutorial and Notebook Version

### PR DESCRIPTION
**Pull request by ddyy345**
Add in tutorial for propane using similar structure as Lennard-Jones tutorial. Made tutorial into notebook format with descriptive comments using opt.py.

### CODE REVIEW

File: msibi/tutorials/README [hunk 1]
```diff
@@ -5,4 +5,9 @@ code works as expected.
 LJ : recover known LJ potential
     - This test, as described in the original paper, is recovering a
       known potential to show that the methodology works. The target
-      data was derived from LJ fluids at three different states. 
+      data was derived from LJ fluids at three different states.
+
+PROPANE : recover known propane potential
+    - This test, as described in the original 2014 paper, is recovering a
+      known potential to show that the methodology works. The target
```
Comment (tcmoore3) [inline msibi/tutorials/README]: Close. We actually are doing coarse-graining here (3-site propane to single-site propane), so this is a bit misleading. You can use section III.C of [this paper](http://aip.scitation.org/doi/abs/10.1063/1.4880555?journalCode=jcp) for reference.

File: msibi/tutorials/propane/opt.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import numpy as np
+
+from msibi import MSIBI, State, Pair, mie
+
+
+# Code below from opt.py for lj tutorial
+# Need for similar code uncertain
+# os.system('rm state*/* rdfs/C3* f_fits.log state*/log.txt')
+# os.system('rm state*/err.txt')
+# os.system('rm state*/query.dcd')
```
Comment (tcmoore3) [inline msibi/tutorials/propane/opt.py]: Remove line 10-11, and either uncomment or delete lines 12-14.

File: msibi/tutorials/propane/opt.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+stateA = State(kT=0.5, state_dir='./state_A', top_file='start.hoomdxml',
+               name='stateA', backup_trajectory=True)
+stateB = State(kT=1.5, state_dir='./state_B', top_file='start.hoomdxml',
+               name='stateB', backup_trajectory=True)
+stateC = State(kT=2.0, state_dir='./state_C', top_file='start.hoomdxml',
+               name='stateC', backup_trajectory=True)
+states = [stateA, stateB, stateC]
+
+# Specify pairs.
+indices = list(itertools.combinations(range(1024), 2))  # all-all for 1468 atoms
```
Comment (tcmoore3) [inline msibi/tutorials/propane/opt.py]: Fix comment.

File: msibi/tutorials/propane/plot.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from msibi_utils.plot_fit import plot_all_fits
+from msibi_utils.plot_rdfs import plot_all_rdfs
+from msibi_utils.animate_rdf import animate_all_pairs_states
+
+plot_all_fits('opt.out', ylims=(0.5, 1))
+plot_all_rdfs('opt.out', 'rdfs', step=4)
+animate_all_pairs_states('opt.out', 'rdfs', step=4, n_skip=0)
```
Comment (tcmoore3) [inline msibi/tutorials/propane/plot.py]: If we're going to include this file, we need to recommend downloading and installing `msibi_utils`.

File: msibi/tutorials/propane/propane.ipynb [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+   "source": [
+    "# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #\n",
+    "# # # # # # WARNING : ENTERING DANGEROUS TERRITORY # # # # # # \n",
+    "# # # # # # # # TRESPASSERS WILL BE SHOT ON SIGHT # # # # # # #\n",
+    "# # # # # # # # # SURVIVORS WILL BE SHOT AGAIN # # # # # # # #\n",
+    "# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #\n",
+    "#\n",
+    "# os.system('rm state*/* rdfs/C3* f_fits.log state*/log.txt')\n",
+    "# os.system('rm state*/err.txt')\n",
+    "# os.system('rm state*/query.dcd')\n",
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: Let's make this a little more subtle; you can make a cell above the "dangerous" cell with a warning.

File: msibi/tutorials/propane/rdfs/fix.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import glob
+import os
+
+for letter in ['A', 'B', 'C']:
+    for file in glob.glob('*state{0}*'.format(letter)):
+        newname = file.replace('state{0}'.format(letter), 'state_{0}'.format(letter))
+        os.system('cp {0} {1}'.format(file, newname))
```
Comment (tcmoore3) [inline msibi/tutorials/propane/rdfs/fix.py]: No need to include this file if we have the rdf names correct in the tutorial.

File: msibi/tutorials/propane/state_A/run.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+
+table.set_from_file('C3', 'C3', filename='/Users/davyyue/Research/msibi/msibi/tutorials/propane/potentials/pot.C3-C3.txt')
+all = hoomd.group.all()
+nvt_int = hoomd.md.integrate.langevin(group=all, kT=T_final, seed=1)
+hoomd.md.integrate.mode_standard(dt=0.001)
+
+hoomd.run(1e2)
+output_dcd = hoomd.dump.dcd(filename='query.dcd', period=100, overwrite=True)
+hoomd.run(1e4)
```
Comment (tcmoore3) [inline msibi/tutorials/propane/state_A/run.py]: This file is generated, so it doesn't need to be in the repository.

File: msibi/tutorials/propane/state_B/run.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+
+table.set_from_file('C3', 'C3', filename='/Users/davyyue/Research/msibi/msibi/tutorials/propane/potentials/pot.C3-C3.txt')
+all = hoomd.group.all()
+nvt_int = hoomd.md.integrate.langevin(group=all, kT=T_final, seed=1)
+hoomd.md.integrate.mode_standard(dt=0.001)
+
+hoomd.run(1e2)
+output_dcd = hoomd.dump.dcd(filename='query.dcd', period=100, overwrite=True)
+hoomd.run(1e4)
```
Comment (tcmoore3) [inline msibi/tutorials/propane/state_B/run.py]: Also don't need this file.

File: msibi/tutorials/propane/state_C/run.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+
+table.set_from_file('C3', 'C3', filename='/Users/davyyue/Research/msibi/msibi/tutorials/propane/potentials/pot.C3-C3.txt')
+all = hoomd.group.all()
+nvt_int = hoomd.md.integrate.langevin(group=all, kT=T_final, seed=1)
+hoomd.md.integrate.mode_standard(dt=0.001)
+
+hoomd.run(1e2)
+output_dcd = hoomd.dump.dcd(filename='query.dcd', period=100, overwrite=True)
+hoomd.run(1e4)
```
Comment (tcmoore3) [inline msibi/tutorials/propane/state_C/run.py]: Also don't need this file.

File: msibi/tutorials/propane/propane.ipynb [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+    "1. standard library imports \n",
+    "2. related third party imports \n",
+    "3. local application/library specific imports\n",
+    "\n",
+    "Group 1 includes `itertools`, `string`, and `os`.\n",
+    "\n",
+    "Group 2 includes `numpy`, which is stored in the keyword `np` and can be accessed with `np.[function]`.\n",
+    "\n",
+    "Group 3 includes `MSIBI`, `State`, `Pair`, and `mie` from the `msibi` package, which manages pair-potential derivation using MultiState Iterative Boltzmann Inversion (MS-IBI).\n"
+   ]
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: You can probably just change this cell to say "Imports," or just leave it out altogether.

File: msibi/tutorials/propane/propane.ipynb [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Specify pairs\n",
+    "\n",
+    "Creates a list of all the possible indices for the 1024 atoms. \n",
+    "\n",
+    "Generates an `initial_guess` for the potential values. \n",
+    "\n",
+    "Uses a list `alphabet` to load each txt rdf file for each state A, B, and C. \n",
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: Not necessary; the code is pretty unambiguous.

File: msibi/tutorials/propane/propane.ipynb [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+    "\n",
+    "Passes the type of interaction analyzed, a `C3` to itself, to `Pair`. Sets the alpha values to `1.0`"
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: Passees the type of interaction "to be optimized"

File: msibi/tutorials/propane/propane.ipynb [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "opt.optimize(states, pairs, n_iterations=5, engine='hoomd')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: Don't need anything below the call to `opt.optimize()`

File: msibi/tutorials/propane/plot.py (hunk 4 above)
Comment (ddyy345) [inline msibi/tutorials/propane/plot.py]: Would this be included in the README file?

File: msibi/tutorials/README (hunk 1 above)
Comment (ddyy345) [inline msibi/tutorials/README]: How would the README for this section be rephrased? How about `This test, as described in the original 2014 paper, performs further optimizations on propane using a coarse-graining 3-1 mapping. The 3-site propane is mapped to the single-site propane and the generated data can be compared to known LJ graphs from literature.`

File: msibi/tutorials/propane/plot.py (hunk 4 above)
Comment (tcmoore3) [inline msibi/tutorials/propane/plot.py]: Yeah, in the README files for the tutorials.

File: msibi/tutorials/README (hunk 1 above)
Comment (tcmoore3) [inline msibi/tutorials/README]: 👍 . And the heading should be something like

PROPANE : optimize pair potentials for a single-site propane model

File: msibi/tutorials/propane/plot.py (hunk 4 above)
Comment (ddyy345) [inline msibi/tutorials/propane/plot.py]: Done with commit yesterday

File: msibi/tutorials/propane/propane.ipynb (hunk 10 above)
Comment (ddyy345) [inline msibi/tutorials/propane/propane.ipynb]: Done with commit yesterday

File: msibi/tutorials/README [hunk 14]
```diff
@@ -1,13 +1,15 @@
 This directory contains several tests.
 It is advisable, at the minimum, to run the LJ test to make sure the
-code works as expected.
+code works as expected. To run plot.py, downloading and installing msibi_utils
+is highly recommended for proper functionality.
```
Comment (tcmoore3) [inline msibi/tutorials/README]: It's not as much for proper functionality, but to visualize the results

File: msibi/tutorials/README [hunk 15]
```diff
@@ -1,13 +1,15 @@
...
 
-PROPANE : recover known propane potential
-    - This test, as described in the original 2014 paper, is recovering a
-      known potential to show that the methodology works. The target
-      data was derived from propane at three different states.
+PROPANE : optimize pair potentials for a single-site propane model
+    - This test, as described in the original 2014 paper, performs further
+      optimizations on propane using a coarse-graining 3-1 mapping. The 3-site
+      propane is mapped to the single-site propane and the generated data can be
+      compared to known LJ graphs from literature.
```
Comment (tcmoore3) [inline msibi/tutorials/README]: "LJ potentials"

And cite the Pu et al paper I cite in that paper.

File: msibi/tutorials/propane/propane.ipynb [hunk 16]
```diff
@@ -9,7 +9,7 @@
     "Created by Davy Yue 2017-06-14\n",
     "\n",
     "\n",
-    "### Import packages\n",
+    "### Imports\n",
     "\n",
     "According to PEP8 standards, imports are grouped into \n",
     "1. standard library imports \n",
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: No need to list pep8 standards in this notebook. The contents of this cell can just be:
"Imports"

File: msibi/tutorials/propane/propane.ipynb [hunk 17]
```diff
@@ -57,18 +58,9 @@
...
-    "#\n",
-    "# os.system('rm state*/* rdfs/C3* f_fits.log state*/log.txt')\n",
-    "# os.system('rm state*/err.txt')\n",
-    "# os.system('rm state*/query.dcd')\n",
-    "#\n",
-    "# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #\n",
-    "# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #"
+    "os.system('rm state*/* rdfs/C3* f_fits.log state*/log.txt')\n",
+    "os.system('rm state*/err.txt')\n",
+    "os.system('rm state*/query.dcd')"
```
Comment (tcmoore3) [inline msibi/tutorials/propane/propane.ipynb]: You can get rid of the "optional...warning:. Just add a note about what it's doing, e.g., "Remove files generated during CG simulations."

File: msibi/tutorials/README (hunk 15 above)
Comment (ddyy345) [inline msibi/tutorials/README]: Should I add in a line after the `PROPANE` section with `Additional citation: P. Liu et al., “Reconstructing atomistic detail for coarse-grained models with resolution exchange,” J. Chem. Phys. 129, 114103 (2008). https://doi.org/10.1063/1.2976663`

**Pull request merged**